src/include/OSL/oslclosure.h

```diff
@@ -118,25 +118,28 @@ struct OSLEXECPUBLIC ClosureColor {
 /// parameter data (mem), but it's expected that the structure be
 /// allocated with enough space to house all the parameter data for
 /// whatever type of custom primitive component it actually is.
-struct OSLEXECPUBLIC ClosureComponent : public ClosureColor
+struct OSLEXECPUBLIC alignas(16) ClosureComponent : public ClosureColor
 {
     Vec3 w;                     ///< Weight of this component
-    alignas(void*) char mem[8]; ///< Memory for the primitive, 8 is the minimum, allocation will be
-                                ///  scaled to requirements of the registered closure struct.
-                                ///  Alignment is forced to a void* to match likely alignement requirements
-                                ///  of the user's structs.
 
-    /// Handy method for getting the parameter memory as a void*.
-    ///
-    void *data () { return &mem; }
-    const void *data () const { return &mem; }
+    /// Handy method for getting the parameter memory as a void*. Allocation
+    /// allocation will be scaled to requirements of the registered closure
+    /// struct. Alignment is forced to 16 bytes to match alignment requirements
+    /// of 64 bit pointers and 128 bit SSE types in user's structs.
+    static const size_t alignment = 16;
+    static size_t aligned_size() {
```

Comment:
I don't think this is needed, right ? If a type has `alignas(N)` then its `sizeof()` will be at least `N` (so that an array of said struct would have properly aligned entries).